Sources/Common/Core/DataArray/index.js

```diff
@@ -36,29 +64,24 @@ function createRangeHelper() {
 }
 
 function computeRange(values, component = 0, numberOfComponents = 1) {
-  const helper = createRangeHelper();
-  const size = values.length;
-  let value = 0;
-
   if (component < 0 && numberOfComponents > 1) {
     // Compute magnitude
+    const size = values.length;
+    let value = 0;
+    const data = [];
```

Comment:
you could allocate the final size of the array instead of "push"

Reply:
I don't quite get how this part is working. Care to send a patch?

Reply:
```
const data = [];
data.length = size / numberOfComponents;
...
  data[i/numberOfComponents] = value;
```
(you may have to have `i` ranging from 0 to size / numberofComponents to reduce the number of operations).
e.g.
```
const numberOfValues = size / numberOfComponents)
const data = new Float64Array(numberOfValues);
for (let i = 0, j = 0; i < numberOfValues; ++i) {
  for (let nextJ = j + numberOfComponents; j < nextJ; ++j) {
     data[i] += values[j]*values[j];
   }
   data[i] *= 0.5;
}
```

Reply:
Couldn't this lead to non-integer numbers as `numberOfValues`?

Reply:
it should never be the case.

Reply:
I pushed the change.

Reply:
Is it any faster ? 
if not, then it may not worth it.

Reply:
Your version is 25% faster